engine/schema/src/main/java/com/cloud/upgrade/dao/Upgrade41500to41510.java

```diff
@@ -235,6 +242,138 @@ public void updateSystemVmTemplates(final Connection conn) {
         LOG.debug("Updating System Vm Template IDs Complete");
     }
 
+    private void correctGuestOsNames(final Connection conn) {
+        guestOsMapper.updateGuestOsName(conn, 7, "Fedora Linux", "Fedora Linux (32 bit)");
+        guestOsMapper.updateGuestOsName(conn, 7, "Mandriva Linux", "Mandriva Linux (32 bit)");
+
+        Ternary<String, String, String> mapping = new Ternary<String, String, String>("VMware", "6.7.3", "opensuseGuest");
+        guestOsMapper.updateGuestOsNameFromMapping(conn, "OpenSUSE Linux (32 bit)", mapping);
+    }
+
+    private void updateGuestOsMappings(final Connection conn) {
+        LOG.debug("Updating guest OS mappings");
+
+        // Add support for SUSE Linux Enterprise Desktop 12 SP3 (64-bit) for Xenserver 8.1.0
+        List<Ternary<String, String, String>> mappings = new ArrayList<Ternary<String, String, String>>();
+        mappings.add(new Ternary<String, String, String>("Xenserver", "8.1.0", "SUSE Linux Enterprise Desktop 12 SP3 (64-bit)"));
+        guestOsMapper.addGuestOsAndHypervisorMappings (conn, 5, "SUSE Linux Enterprise Desktop 12 SP3 (64-bit)", mappings);
+        mappings.clear();
+
+        // Add support for SUSE Linux Enterprise Desktop 12 SP4 (64-bit) for Xenserver 8.1.0
+        mappings.add(new Ternary<String, String, String>("Xenserver", "8.1.0", "SUSE Linux Enterprise Desktop 12 SP4 (64-bit)"));
+        guestOsMapper.addGuestOsAndHypervisorMappings (conn, 5, "SUSE Linux Enterprise Desktop 12 SP4 (64-bit)", mappings);
+        mappings.clear();
+
+        // Add support for SUSE Linux Enterprise Server 12 SP4 (64-bit) for Xenserver 8.1.0
+        mappings.add(new Ternary<String, String, String>("Xenserver", "8.1.0", "SUSE Linux Enterprise Server 12 SP4 (64-bit)"));
+        mappings.add(new Ternary<String, String, String>("Xenserver", "8.1.0", "NeoKylin Linux Server 7"));
+        guestOsMapper.addGuestOsAndHypervisorMappings(conn, 5, "SUSE Linux Enterprise Server 12 SP4 (64-bit)", mappings);
+        mappings.clear();
+
+        // Add support for Scientific Linux 7 for Xenserver 8.1.0
+        mappings.add(new Ternary<String, String, String>("Xenserver", "8.1.0", "Scientific Linux 7"));
+        guestOsMapper.addGuestOsAndHypervisorMappings (conn, 9, "Scientific Linux 7", mappings);
+        mappings.clear();
+
+        // Add support for NeoKylin Linux Server 7 for Xenserver 8.1.0
+        guestOsMapper.addGuestOsAndHypervisorMappings(conn, 9, "NeoKylin Linux Server 7", mappings); //334
+        mappings.clear();
+
+        // Pass Guest OS Ids to update pre-4.14 mappings
+        // Add support CentOS 8 for Xenserver 8.1.0
```

Comment:
Please note, the hypervisor mapping for the guest OSes added pre-4.14 (with hardcoded ids < 305), are updated using the same ids.